### PR TITLE
Fixed bug that causes weights to be not updated

### PR DESCRIPTION
**Pull request by theo1701**
I think by returning the updated weight, it also improves code readability.

### CODE REVIEW

File: python/omnifold.py [hunk 1]
```diff
@@ -212,16 +212,13 @@ def omnifold(
...
-# weights_push is modified for reweighting
-# Return: None
 def train_step(
     # Data
     X_step, # feature array of observed data, for this step
     Y_step, # label array, for this step
     w_step, # w_array, for this step
-    weights_update, # arraylike, weights_pull or weights_push to be updated
-    weights_update_range, # range of weights_update to be updated, the entire weights_update_range is updated if NONE is supplied
+    weights_multiplier, # arraylike, multiplied by the model reweight to produce the new weight to be returned
```
Comment (zctao) [inline python/omnifold.py:220]: I don't think It is necessary to have `weights_multiplier` as an argument. The only time it is not `None` is at step 1, and the factor can probably be applied on the returned value of the function: `weights_pull = weights_pull * train_step(...)`. Without `weights_multiplier`, the nested ifs for return can also be simplified for better readability.

File: python/omnifold.py [hunk 2]
```diff
@@ -212,16 +212,13 @@ def omnifold(
 
     return weights_unfold
 
-# evaluate a step in OmniFold method, training a model if needed and train the model
-# weights_push is modified for reweighting
-# Return: None
 def train_step(
```
Comment (zctao) [inline python/omnifold.py:215]: Sorry for asking again about the function name, but with the updates, maybe `reweight_step` would be a more suitable name?

File: python/omnifold.py (hunk 1 above)
Comment (theo1701) [reply python/omnifold.py:220]: Sure!

File: python/omnifold.py (hunk 2 above)
Comment (theo1701) [reply python/omnifold.py:215]: Yes you are right. The original plan was to handle all training and weight updating in one function. Now the main purpose of the function is to train the model and return the updated weights. How about get_step_reweight?

**Comment by zctao** [comment]
Should I merge this now? Or you'd like to test more?

**Comment by theo1701** [comment]
I think it can be merged now.

**Pull request merged**